### PR TITLE
fix(query): Need return error udf_name when derialize from pb  failed.

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

https://github.com/datafuselabs/databend/pull/12729/files#diff-9c992028e59caebc313d761b8488b17f142618fb89db64c51c1655689d68c41b

image

I think we can not upgrade the old json to pb. The best way is return error udf when `select * from system.user_functions` and then user need to re-create the udf.

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [x] No Test  - rollback deserialize udf code. If the test can be passed the code is ok.

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [x] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14964)

### CODE REVIEW

**Review by drmingdrmer — COMMENTED** [review]
What does this PR do?

It reverted to using prefix_list_kv() to retrieve from meta-service, that's all?

> Reviewed 2 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14964#-:-Nt0CiSh9G_lAoa3aeKp:bf4dz8i)* status: 2 of 4 files reviewed, all discussions resolved (waiting on @sundy-li)

**Comment by TCeason** [comment]
> > What does this PR do?
> > It reverted to using prefix_list_kv() to retrieve from meta-service, that's all?
> > > Reviewed 2 of 4 files at r1, all commit messages.
> > > _[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14964#-:-Nt0CiSh9G_lAoa3aeKp:bf4dz8i)_ status: 2 of 4 files reviewed, all discussions resolved (waiting on @sundy-li)
> 
> just return a error msg with udf_name when deserialize from pb error.
> 
> Then user needs to re-create the udf.

cc @BohuTANG How about this case?

File: src/query/management/src/udf/udf_mgr.rs [hunk 1]
```diff
@@ -121,16 +121,24 @@ impl UdfMgr {
     /// Get all the UDFs for a tenant.
     #[async_backtrace::framed]
     #[minitrace::trace]
-    pub async fn list_udf(&self) -> Result<Vec<UserDefinedFunction>, UdfApiError> {
```
Comment (sundy-li) [inline src/query/management/src/udf/udf_mgr.rs:124]: maybe we add another function named `list_udf_fallback` when the error occurs.

Comment (TCeason) [reply src/query/management/src/udf/udf_mgr.rs:124]: Ok, it is safer to deal only with error cases. Please review the newest commit.

**Review by sundy-li — APPROVED** [review]

**Review by sundy-li — APPROVED** [review]

**Pull request merged**